queue/driver_mongo.go

```diff
@@ -399,6 +421,30 @@ func (d *mongoDriver) processJobForGroup(j *registry.JobInterchange) {
 	j.Name = buildCompoundID(d.opts.GroupName, j.Name)
 }
 
+func (d *mongoDriver) removeGroupFromMetadata(j *registry.JobInterchange) {
+	if !d.opts.UseGroups {
+		return
+	}
+
+	j.Name = deconstructCompoundID(j.Name, d.opts.GroupName)
+}
+
+func (d *mongoDriver) addRetryToMetadata(j *registry.JobInterchange) {
+	if !j.RetryInfo.Retryable {
+		return
+	}
+
+	j.Name = buildCompoundID(retryAttemptPrefix(j.RetryInfo.CurrentAttempt), j.Name)
+}
+
+func (d *mongoDriver) removeRetryFromMetadata(j *registry.JobInterchange) {
+	if !j.RetryInfo.Retryable {
+		return
+	}
+
+	j.Name = deconstructCompoundID(j.Name, retryAttemptPrefix(j.RetryInfo.CurrentAttempt))
```

Comment:
This shouldn't be possible because `removeMetadata()` is only used on jobs deserialized from the DB, which means that the only way someone could break this would be by changing the driver to modify the attempt. This method is only necessary for `JobStats` to return the job ID without the retry/group info.